packages/core/useConfirmDialog/index.ts

```diff
@@ -0,0 +1,73 @@
+import { Ref } from 'vue-demi'
+import { createEventHook, EventHook, Fn } from '@vueuse/core'
+
+export interface useConfirmDialogReturn<T, D, E> {
+  /*
+  * Opens the dialog.
+  */
+  showDialog: Fn
```

Comment:
I would like to have this named `reveal` or `prompt` and have it return a Promise that resolves on dialog closed. Something like

```ts
const { data, isCanceled } = await dialog.reveal()
```

Reply:
Oh, I forgot to mention this in the interface, `showDialog` passes data from its attribute to the `onShowDialog` hook. I also thought about the promise, but in the end, I concluded that hooks completely replace the promise, and save us from the callback hell.

Reply:
I would find it more useful to await the dialog and get the payload from it so I could continue the process. We could have them both.

Reply:
Ok, maybe it will be more useful for those who like to work with promises. Then I will add it, by the way, it was my first version of this function.

Reply:
I'll do it tomorrow, it's too many changes in all files.

Reply:
I'm with @antfu here 👍 this would be a nice addition